contracts/CrowdsaleExt.sol

```diff
@@ -436,23 +439,22 @@ contract CrowdsaleExt is Haltable {
 
   /**
    * Allow addresses to do early participation.
-   *
-   * TODO: Fix spelling error in the name
    */
-  function setEarlyParicipantWhitelist(address addr, bool status, uint minCap, uint maxCap) onlyOwner {
+  function setEarlyParticipantWhitelist(address addr, bool status, uint minCap, uint maxCap) onlyOwner {
     if (!isWhiteListed) throw;
     earlyParticipantWhitelist[addr] = WhiteListData({status:status, minCap:minCap, maxCap:maxCap});
+    whitelistedParticipants.push(addr);
```

Comment:
Note that this does not check if the address was already added. Should we do that to avoid duplicates or is a lower gas usage more important?

Reply:
I think, we could simply check, that earlyParticipantWhitelist[addr].maxCap == 0:
```
function setEarlyParticipantWhitelist(address addr, bool status, uint minCap, uint maxCap) onlyOwner {
        if (!isWhiteListed) throw;
        if (earlyParticipantWhitelist[addr].maxCap == 0) {
            earlyParticipantWhitelist[addr] = WhiteListData({status:status, minCap:minCap, maxCap:maxCap});
            whitelistedParticipants.push(addr);
        }
    }
```
I've checked, that without this check transaction costs **94868** gas, and with it: **95190** gas. It isn't so much.
Also, it would be nice, if we'll add these checks:
```
assert(addr != address(0));
assert(minCap <= maxCap);
```

Reply:
I'm adding this, but I have a minor concern: you could call the method twice with 0 `minCap` and `maxCap` and it will enter the if the two times. I guess we could add another assert `assert(maxCap > 0)`. Do you think it's worth it?

Reply:
I just added the changes that you proposed, and the proper tests. Let me know if you also want me to add the check that maxCap is positive.

Reply:
@fvictorio sorry, Franco, I missed your comment. Yes, let's add a check, that cap values are positive. But, maybe we should check, that minCap is positive (not maxCap), because we've already checked, that maxCap >= minCap?